tests/tests/tys.rs

```
@@ -21,18 +21,53 @@ macro_rules! def_num_ty_tests {
+                            val_as_u64 <= max_unsigned
+                        });
+                    }
+                }
 
                 // Test 1: All test values round-trip
-                for &val in test_values {
+                for &val in &test_values {
                     let created = Foo::create().val(val).exec(&db).await.unwrap();
                     let read = Foo::get_by_id(&db, &created.id).await.unwrap();
                     assert_eq!(read.val, val, "Round-trip failed for: {}", val);
+
+                    // Raw storage verification - verify the value is stored correctly at the database level
+                    let mut filter = std::collections::HashMap::new();
+                    filter.insert("id".to_string(), toasty_core::stmt::Value::from(created.id));
+
+                    match s.get_raw_column_value::<$t>("foos", "val", filter).await {
+                        Ok(raw_stored_value) => {
+                            assert_eq!(
+                                raw_stored_value, val,
+                                "Raw storage verification failed for {}: expected {}, got {}",
+                                stringify!($t), val, raw_stored_value
+                            );
+                        }
+                        Err(e) => {
+                            let error_msg = format!("{}", e);
+                            if !(error_msg.contains("not yet implemented") ||
+                                 error_msg.contains("relation") && error_msg.contains("does not exist") ||
+                                 error_msg.contains("NUMERIC type conversion not yet implemented")) {
+                                // Only panic on unexpected errors
+                                panic!("Unexpected raw storage error for {} value {}: {}", stringify!($t), val, error_msg);
+                            }
+                            // Expected errors are silently ignored
+                        }
```

Always panic if get_raw_column_value returns Err. We don't want silent errors.